src/core/lightning/index.ts

```diff
@@ -201,19 +203,27 @@ export const LightningMixin = (superclass) =>
         params,
       })
 
-      const remainingTwoFALimit = await this.user.remainingTwoFALimit()
+      const twoFALimitsChecker = await getLimitsChecker(this.user.id)
+      if (twoFALimitsChecker instanceof Error) throw twoFALimitsChecker
 
-      if (this.user.twoFA.secret && remainingTwoFALimit < tokens) {
-        if (!twoFAToken) {
-          throw new TwoFAError("Need a 2FA code to proceed with the payment", {
-            logger: lightningLogger,
-          })
-        }
+      const user = await UsersRepository().findById(this.user.id)
+      if (user instanceof Error) throw user
+      const { twoFA } = user
 
-        if (!verifyToken(this.user.twoFA.secret, twoFAToken)) {
-          throw new TwoFAError(undefined, { logger: lightningLogger })
-        }
-      }
+      const twoFACheck = twoFA?.secret
+        ? await checkTwoFA({
+            amount: toSats(tokens),
+            twoFAToken: twoFAToken as TwoFAToken,
```

Comment:
`towFAToken` can be undefined, the casting here hides that from the type system

Reply:
Fixed: https://github.com/GaloyMoney/galoy/pull/583/commits/5b24d043aa744a41b3f6343a1d5baa8d9a2cbbd6